src/include/packager/repositories_include.rb

```diff
@@ -25,329 +34,179 @@ def initialize_packager_repositories_include(_include_target)
+      if !service_type.nil? && service_type != "NONE"
+        add_service(url, preffered_name)
+        return :ok
+      end
 
-            if url_path.nil? || url_path == ""
-              url_path = Ops.get(
-                p_elems,
-                Ops.subtract(Builtins.size(p_elems), 2),
-                fallback
-              )
+      found_products = scan_products(expanded_url, url)
+      newSources = []
 
-              url_path = fallback if url_path.nil? || url_path == ""
-            end
-          end
+      enter_again = false
 
-          new_repos = [[url_path, "/"]]
-        end
+      # more products on the medium, let the user choose the products to install
+      # this code is not used in AutoYaST, but rather be on the safe side...
+      if !Mode.auto && found_products.size > 1
+        require "y2packager/dialogs/addon_selector"
+        dialog = Y2Packager::Dialogs::AddonSelector.new(found_products)
 
-        newSources = []
+        ui = dialog.run
+        found_products = dialog.selected_products
 
-        enter_again = false
-
-        Builtins.foreach(new_repos) do |repo|
-          next if enter_again
-          prod_dir = Ops.get(repo, 1, "/")
-          name = Ops.get(repo, 0, "")
-          if !preffered_name.nil? && preffered_name != ""
-            name = preffered_name
-            name += " (#{prod_dir})" if ![nil, "", "/"].include?(prod_dir)
-          end
-          # probe repository type (do not probe plaindir repo)
-          repo_type = plaindir ? @plaindir_type : Pkg.RepositoryProbe(expanded_url, prod_dir)
-          Builtins.y2milestone(
-            "Repository type (%1,%2): %3",
-            URL.HidePassword(url),
-            prod_dir,
-            repo_type
-          )
-          # the probing has failed
-          if repo_type.nil? || repo_type == "NONE"
-            if scheme == "dir"
-              if !Popup.AnyQuestion(
-                Popup.NoHeadline,
-                # continue-back popup
-                _(
-                  "There is no product information available at the given location.\n" \
-                    "If you expected to to point a product, go back and enter\n" \
-                    "the correct location.\n" \
-                    "To make rpm packages located at the specified location available\n" \
-                    "in the packages selection, continue.\n"
-                ),
-                Label.ContinueButton,
-                Label.BackButton,
-                :focus_no
-              )
-                enter_again = true
-                next
-              end
+        # nothing selected or pressed abort/cancel/close/back/...
+        return :cancel if found_products.empty? || ui != :next
+      end
 
-              repo_type = @plaindir_type
-              Builtins.y2warning(
-                "Probing has failed, using Plaindir repository type."
-              )
+      found_products.each do |product|
+        next if enter_again
```

Comment:
this smells. What is logic there and why break caant be used at line 119?

Reply:
Um, this would need more analysis how the code is called from outside, although it looks strange I do not date to touch this now...